spot_wrapper/wrapper.py

```diff
@@ -648,255 +439,298 @@ def __init__(
+                self._graph_nav_client = self._robot.ensure_client(
+                    GraphNavClient.default_service_name
+                )
+                self._map_processing_client = self._robot.ensure_client(
+                    MapProcessingServiceClient.default_service_name
+                )
+                self._power_client = self._robot.ensure_client(
+                    PowerClient.default_service_name
+                )
+                self._lease_client = self._robot.ensure_client(
+                    LeaseClient.default_service_name
+                )
+                self._lease_wallet = self._lease_client.lease_wallet
+                self._image_client = self._robot.ensure_client(
+                    ImageClient.default_service_name
+                )
+                self._estop_client = self._robot.ensure_client(
+                    EstopClient.default_service_name
+                )
+                self._docking_client = self._robot.ensure_client(
+                    DockingClient.default_service_name
+                )
+                self._spot_check_client = self._robot.ensure_client(
+                    SpotCheckClient.default_service_name
+                )
+                if self._robot.has_arm():
+                    self._manipulation_client = self._robot.ensure_client(
+                        ManipulationApiClient.default_service_name
                     )
-                    if self._robot.has_arm():
-                        self._manipulation_client = self._robot.ensure_client(
-                            ManipulationApiClient.default_service_name
-                        )
-                    initialised = True
-                except Exception as e:
-                    sleep_secs = 15
-                    self._logger.warning(
-                        "Unable to create client service: {}. This usually means the robot hasn't "
-                        "finished booting yet. Will wait {} seconds and try again.".format(
-                            e, sleep_secs
-                        )
+
+                self._robot_clients = {
+                    "robot_state_client": self._robot_state_client,
+                    "robot_command_client": self._robot_command_client,
+                    "graph_nav_client": self._graph_nav_client,
+                    "map_processing_client": self._map_processing_client,
+                    "power_client": self._power_client,
+                    "lease_client": self._lease_client,
+                    "image_client": self._image_client,
+                    "estop_client": self._estop_client,
+                    "docking_client": self._docking_client,
+                    "spot_check_client": self._spot_check_client,
+                    "robot_command_method": self._robot_command,
+                    "world_objects_client": self._world_objects_client,
+                    "manipulation_client": self._manipulation_client,
+                }
```

Comment:
This makes me think that we should implement something like a client manager which can be used to access these things, and probably initialise them as well. But that's probably for later rather than in this PR.

Reply:
Agreed, this works for now (since almost everything is an object in Python), but it would be nice to be explicit about what needs to be defined at runtime before going to operation.